vyper/semantics/validation/annotation.py

```diff
@@ -65,11 +66,13 @@ def visit_Assert(self, node):
 
     def visit_Assign(self, node):
         type_ = get_exact_type_from_node(node.target)
+        node._metadata["type"] = type_
```

Comment:
why would `Assign` have a type?

Reply:
I added it as an extension to the request in 2276 for more information on the type of `AnnAssign` nodes (variable declarations) since they are similar in that the variable type is usually derived from the child nodes.

Reply:
i don't think 2276 was literally asking for vyper type info on `AnnAssign` nodes :) i think they were more asking "what kind of declaration is this". it doesn't make really make sense for statements to have types attached to them -- types generally go on expressions.

Reply:
Ah okay got it, I will revert the changes made to the various `*Assign` nodes.